### ISSUE TITLE
API V2.1.0 - Machine readable tags or IDs

### ISSUE THREAD

**Issue by srtalbot** [cds-snc/platform-forms-client#4378]

**Comment by connorscarolyns** [comment]
@thiessenp-cds @timarney we'll definitely need to talk this one through together to better understand the problems we are trying to solve. Do you think we should include one of our prototype partners who expressed this need to get a better idea of their needs?

**Comment by Abi-Nada** [comment]
Connect with @bryan-robitaille and @craigzour first on ideas to get the ball rolling

**Comment by connorscarolyns** [comment]
Proposed approach:

Phase 1 - 
Made sure current question IDs are static and consistent when forms are downloaded and re-uploaded

Phase 2- 
Create a custom editable tag that users can add to their questions to enable consistency between different forms that use the same field types, like "Name"

@srtalbot @timarney @bryan-robitaille @Abi-Nada

**Comment by connorscarolyns** [comment]
@samsadasivan will be consulted on design when interface changes are ready.

**Comment by connorscarolyns** [comment]
@Abi-Nada should we split this into two tickets? one for the static ID work and one for the editable tags?

**Comment by timarney** [comment]
I think we should look at the `editable tags` first as that might help guide the path for the static ID work.

Looking to outline both before we make schema updates.

cc: @samsadasivan @connorscarolyns

**Comment by anikbrazeau** [comment]
Figma: https://www.figma.com/design/zlh8BVQyDczHvXo4xXUrb6/GC-Forms-API-interface?node-id=4766-37281&t=vowAaPR0FqR7LOHp-4

**Comment by timarney** [comment]
- [ ] update element more menu to allow tag input (default to uuid)
- [ ] update template store to include tag / uuid


<hr>

- [ ] update increment id to store "next id"
- [ ] update template store to use "next id"


<hr>

- [ ]  test with repeating sets
- [ ] test with address complete

**Comment by dsamojlenko** [comment]
Work is underway here: https://github.com/cds-snc/platform-forms-client/pull/4694

**Comment by connorscarolyns** [comment]
One use case I'm aware of is to provide a way to standardize questions across forms so that if different forms are going into one system, all first names, last names, and phone numbers can be scripted once to land in the right place without having to rewire each one individually.

**Comment by anikbrazeau** [comment]
As a form builder, I want to change the form question content and the order of the form without breaking the API integration, so that I can keep tinkering with the form even once the API is set up.

As a developer, I want to set up the form response delivery mechanism via API, without having to overhaul it every time the form changes slightly.

As a client, I want a logical default option for consistent tags, so that I don't have to set tags or keys up for everything manually.

**Comment by anikbrazeau** [comment]
**Next focus:** validation and error handling
 - cannot be empty
 - ensure ID uniqueness
 - duplicate tags
 - invalid format
   - character limits
   - types of characters
   - length
   - amount

**Comment by anikbrazeau** [comment]
PR: #5448
Ready for QA testing this week
